modules/custom/dkan_api/src/Controller/Api.php

```diff
@@ -3,38 +3,81 @@
 namespace Drupal\dkan_api\Controller;
 
 use Drupal\Core\Controller\ControllerBase;
-use Symfony\Component\HttpFoundation\JsonResponse;
 use Sae\Sae;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
+/**
+ *
+ */
 abstract class Api extends ControllerBase {
 
+  /**
+   * Drupal service container.
+   *
+   * @var \Symfony\Component\DependencyInjection\ContainerInterface
+   */
+  protected $container;
+  
+  /**
+   * Factory to generate various dkan classes.
+   * 
+   * @var \Drupal\dkan_common\Service\Factory 
+   */
+  protected $dkanFactory;
+
+  /**
+   *
+   */
+  public function __construct(ContainerInterface $container) {
+    $this->container = $container;
+    $this->dkanFactory = $container->get('dkan.factory');
+  }
+
+  /**
+   *
+   */
   abstract protected function getJsonSchema();
 
+  /**
+   *
+   */
   abstract protected function getStorage();
 
+  /**
+   *
+   */
   public function getAll() {
     /* @var $engine \Sae\Sae */
     $engine = $this->getEngine();
 
     $datasets = $engine->get();
     $json_string = "[" . implode(",", $datasets) . "]";
```

Comment:
not sure what the resoning behind this line is or how to refactor. It seems to be manually creating a json string instead of using json_encode. but it is followed by a json_decode immediately on next line.

Reply:
@ketwaroo $datasets is an array of json strings, so json_encode will not do the right thing.